### PR TITLE
fix #80: alpha pending parses repo path

### LINKED ISSUE #80: github.issue_update_alpha_pending needs to parse repo along with issue
radiasoft/download#NNN needs to be parsed

### PR DESCRIPTION
**Pull request by rorour**

### CODE REVIEW

**Comment by e-carlin** [comment]
@rorour I think we now have access to use the test-pykern-github repo.

You should be able to run the test with `PYKERN_PKCLI_GITHUB_USER='rorour' PYKERN_PKCLI_GITHUB_PASSWORD=$GITHUB_PAT pykern test tests/pkcli/github_test.py`

It looks like in the non full_path case the repo name is being added to the issue. For example this issue https://github.com/radiasoft/test-pykern-github/issues/196 has `EsZDH9wxwzv91Fzy for github_test test-pykern-github#192` and I think that line should be just `EsZDH9wxwzv91Fzy for github_test #192`. The full_path case looks good tho!

**Review by e-carlin — COMMENTED** [review]
Woops had these comments too but forgot to submit my review. Sorry

File: tests/pkcli/github_test.py [hunk 1]
```diff
@@ -89,5 +95,10 @@ def _create_commit(repo):
     x = re.sub(r'(?<=github_test)=([^\n]+)', t, b)
     if b == x:
         b += f'github_test={t}\n'
-    m.update(f'fix #{i.number} for github_test', pkcompat.to_bytes(b))
-    return (f'#{i.number}', t)
+    if full_path:
```
[the remark this answers is not in the document]
Comment (rorour) [reply tests/pkcli/github_test.py]: e51ed5ad795072ccb5951e6c909230393ac6dd6c

File: tests/pkcli/github_test.py [hunk 2]
```diff
@@ -49,6 +49,12 @@ def test_issue_start():
     a = github.issue_update_alpha_pending(github._TEST_REPO)
     pkunit.pkre(i, a)
     pkunit.pkre(t, a)
+
```
[the remark this answers is not in the document]
Comment (rorour) [reply tests/pkcli/github_test.py]: e51ed5ad795072ccb5951e6c909230393ac6dd6c

**Comment by e-carlin** [comment]
@rorour I made some updates https://github.com/radiasoft/pykern/pull/119/commits/603cb9751d3d13d71b3d9884554a248197798b94 . Let me know if you see any problems. I reworked the regex a bit but my regex skills still need work so maybe I'm missing something yours was covering. Also, I think it simplified the logic to just always use the repo full_name when tagging the issue in the release issue.

File: tests/pkcli/github_test.py [hunk 3]
```diff
@@ -45,18 +45,20 @@ def test_issue_start():
     github.issue_pending_alpha(github._TEST_REPO)
     with pkunit.pkexcept('prior release'):
         github.issue_start_alpha(github._TEST_REPO)
-    i, t = _create_commit(r)
-    a = github.issue_update_alpha_pending(github._TEST_REPO)
-    pkunit.pkre(i, a)
-    pkunit.pkre(t, a)
+    for b in (False, True):
+        i, t = _create_commit(r, full_name=b)
+        f = f'{r.full_name}#{i}'
```
Comment (robnagler) [inline tests/pkcli/github_test.py:50]: You want to test both cases, and this only tests, one. We want to see that both issues got added.

This is when I would make it a nested function, something like (see changes below):

```py
def _commit(repo, full_name, issues):
    issues.append(_create_commit(repo, full_name=full_name))
    a = github.issue_update_alpha_pending(github._TEST_REPO)
    pkunit.pkre(issues[-1].title, a)
    pkunit.pkre(issues[-1].link, a)
    
issues = []
_commit(r, False, issues)
_commit(r, True, issues)
```

Then `issues` will be used below.

File: tests/pkcli/github_test.py [hunk 4]
```diff
@@ -45,18 +45,20 @@ def test_issue_start():
...
+        pkunit.pkre(f, a)
     m = None
     for x in (
         github.issue_start_alpha,
         github.issue_start_beta,
         github.issue_start_prod,
     ):
         a = x(github._TEST_REPO)
-        pkunit.pkre(i, a)
+        pkunit.pkre(f, a)
```
Comment (robnagler) [inline tests/pkcli/github_test.py:61]:
```py
for i in issues:
     # approximate match, but there won't be false positives
     pkunit.pkre(i.link, a)
     pkunit.pkre(i.title, a)
```

File: tests/pkcli/github_test.py [hunk 5]
```diff
@@ -89,5 +91,6 @@ def _create_commit(repo):
     x = re.sub(r'(?<=github_test)=([^\n]+)', t, b)
     if b == x:
         b += f'github_test={t}\n'
-    m.update(f'fix #{i.number} for github_test', pkcompat.to_bytes(b))
-    return (f'#{i.number}', t)
+    r = f'{repo if full_name else ""}#{i.number}'
+    m.update(f'fix {r} for github_test', pkcompat.to_bytes(b))
+    return (i.number, t)
```
Comment (robnagler) [inline tests/pkcli/github_test.py:96]:
```py
return PKDict(link=f'{repo.full_name}#{i}', title=t)
```

**Review by robnagler — CHANGES_REQUESTED** [review]

**Comment by robnagler** [comment]
What's the status of this?

**Comment by e-carlin** [comment]
We worked on it today. I need to review and then I'll loop you in for final review.

**Review by e-carlin — APPROVED** [review]

**Comment by e-carlin** [comment]
LGTM. @robnagler please give a final review.

**Review by robnagler — APPROVED** [review]

**Pull request merged**